src/common/service/NetRemoteService.cxx

```diff
@@ -316,6 +316,19 @@ IAccessPointWeakToNetRemoteAccessPointResultItem(std::weak_ptr<Microsoft::Net::W
     return item;
 }
 
+std::unique_ptr<Microsoft::Net::Wifi::IAccessPointController>
+IAccessPointWeakToAccessPointController(std::weak_ptr<Microsoft::Net::Wifi::IAccessPoint>& accessPointWeak)
+{
+    auto accessPoint = accessPointWeak.lock();
+    if (accessPoint != nullptr) {
+        return accessPoint->CreateController();
+    } else {
+        LOGE << "Failed to retrieve access point";

```

Comment:
Adding why it could not be retrieved would make decoding what happened from the logs a bit easier:

```suggestion
        LOGE << "Failed to retrieve access point as it is no longer valid";
```